resources/js/Pages/Galleries/hooks/useWalletDraftGalleries.ts

```diff
@@ -0,0 +1,218 @@
+import { useCallback, useEffect, useState } from "react";
+import { useIndexedDB } from "react-indexed-db-hook";
+import { isTruthy } from "@/Utils/is-truthy";
+
+const MAX_DRAFT_LIMIT_PER_WALLET = 6;
+const DRAFT_TTL_DAYS = 30;
+
+interface Properties {
+    address: string;
+    limit?: number;
+}
+
+export interface DraftNft {
+    nftId: number;
+    image: string;
+    collectionSlug: string;
+}
+
+export interface GalleryDraft {
+    title: string;
+    cover: ArrayBuffer | null;
+    coverType: string | null;
+    nfts: DraftNft[];
+    walletAddress?: string;
+    id?: number | null;
+    value: string | null;
+    collectionsCount: number;
+    updatedAt: number | null;
+}
+
+interface WalletDraftGalleriesState {
+    upsert: (draft: GalleryDraft) => Promise<GalleryDraft>;
+    remove: (id?: number | null) => Promise<void>;
+    removeExpired: () => Promise<void>;
+    drafts: GalleryDraft[];
+    findById: (id: number | string) => Promise<GalleryDraft | undefined>;
+    isLoading: boolean;
+    isSaving: boolean;
+    hasReachedLimit: boolean;
+}
+
+export const useWalletDraftGalleries = ({ address }: Properties): WalletDraftGalleriesState => {
+    const database = useIndexedDB("gallery-drafts");
+    const [drafts, setDrafts] = useState<GalleryDraft[]>([]);
+    const [isLoading, setIsLoading] = useState(true);
+    const [isSaving, setIsSaving] = useState(false);
+    const [hasReachedLimit, setHasReachedLimit] = useState(false);
+
+    const updateDraftState = useCallback(async () => {
+        setIsLoading(true);
+
+        const drafts = await allDrafts();
+
+        setDrafts(drafts);
+        setIsLoading(false);
+
+        setHasReachedLimit(drafts.length >= MAX_DRAFT_LIMIT_PER_WALLET);
+    }, [address]);
+
+    useEffect(() => {
+        void updateDraftState();
+    }, [address]);
+
+    /**
+     * Add new draft gallery.
+     *
+     * @param {GalleryDraft} draft
+     * @returns {Promise<GalleryDraft>}
+     */
+    const add = async (draft: GalleryDraft): Promise<GalleryDraft> => {
+        const { id: _, ...draftToSave } = draft;
+        const drafts = await allDrafts();
+
+        if (drafts.length >= MAX_DRAFT_LIMIT_PER_WALLET) {
+            throw new Error("[useWalletDraftGalleries:add] Limit Reached");
```

Comment:
Do you think we need to throw an error here? This will enforce us to wrap this call in `try-catch` statement which we could return undefined instead handle limit thing gracefully.